From 105d279a8f1b7ec3679d93a7c6c1a59ffa9bbcba Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Sat, 22 Aug 2020 22:51:03 +0200
Subject: [PATCH] ARM: dts: stm32: Add PDK2 DTO for DH 497-200 with
 DataImage SCF0700C48GGU06CTP panel

Add DT overlay for the PDK2 board for DH 497-200 adapter card with
DataImage SCF0700C48GGU06CTP Parallel RGB panel attached to it. The
Linux DT compatible string matches on SCF0700C48GGU18, which is a
compatible panel, just like FG0700G3DSSWAGT1 and FG0700G3DSSWBG01.

Upstream-Status: Inappropriate
Signed-off-by: Marek Vasut <marex@denx.de>
---
 arch/arm/boot/dts/Makefile                    |  1 +
 ...2mp15xx-dhcom-pdk2-overlay-497-200-x12.dts | 66 +++++++++++++++++++
 2 files changed, 67 insertions(+)
 create mode 100644 arch/arm/boot/dts/stm32mp15xx-dhcom-pdk2-overlay-497-200-x12.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index c82ceee7ef7f..7c32bd8e007e 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -1004,6 +1004,7 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157a-dk1.dtb \
 	stm32mp157c-dhcom-pdk2.dtb \
 	stm32mp15xx-dhcom-pdk2-overlay-460-200-x11.dtbo \
+	stm32mp15xx-dhcom-pdk2-overlay-497-200-x12.dtbo \
 	stm32mp15xx-dhcom-pdk2-overlay-531-100-x21.dtbo \
 	stm32mp15xx-dhcom-pdk2-overlay-531-100-x22.dtbo \
 	stm32mp15xx-dhcom-pdk2-overlay-505-200-x12-ch101olhlwh.dtbo \
diff --git a/arch/arm/boot/dts/stm32mp15xx-dhcom-pdk2-overlay-497-200-x12.dts b/arch/arm/boot/dts/stm32mp15xx-dhcom-pdk2-overlay-497-200-x12.dts
new file mode 100644
index 000000000000..5764cb2e62be
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp15xx-dhcom-pdk2-overlay-497-200-x12.dts
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) 2020 Marek Vasut <marex@denx.de>
+ */
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/interrupt-controller/arm-gic.h>
+#include <dt-bindings/pwm/pwm.h>
+
+/dts-v1/;
+/plugin/;
+
+&{/} {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	display_bl: display-bl {
+		compatible = "pwm-backlight";
+		pwms = <&pwm2 3 500000 PWM_POLARITY_INVERTED>;
+		brightness-levels = <0 16 22 30 40 55 75 102 138 188 255>;
+		default-brightness-level = <8>;
+		enable-gpios = <&gpioi 0 GPIO_ACTIVE_HIGH>;
+		status = "okay";
+	};
+
+	panel {
+		compatible = "dataimage,scf0700c48ggu18";
+		backlight = <&display_bl>;
+
+		port {
+			lcd_panel_in: endpoint {
+				remote-endpoint = <&lcd_display_out>;
+			};
+		};
+	};
+};
+
+&timers2 {
+	/* spare dmas for other usage (un-delete to enable pwm capture) */
+	dmas;
+	dma-names;
+	status = "okay";
+
+	pwm2: pwm {
+		#pwm-cells = <3>;
+		pinctrl-0 = <&pwm2_pins_a>;
+		pinctrl-names = "default";
+		status = "okay";
+	};
+
+	timer@1 {
+		status = "okay";
+	};
+};
+
+&ltdc {
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&ltdc_pins_b>;
+	pinctrl-1 = <&ltdc_sleep_pins_b>;
+	status = "okay";
+
+	port {
+		lcd_display_out: endpoint {
+			remote-endpoint = <&lcd_panel_in>;
+		};
+	};
+};
-- 
2.28.0

